src/core/orderedrelationmodel.cpp

```diff
@@ -0,0 +1,195 @@
+
+#include "orderedrelationmodel.h"
+#include "fieldexpressionvaluesgatherer.h"
+#include "qgsrelation.h"
+#include "qgsfeature.h"
+#include "qgsfeaturerequest.h"
+#include "qgsvectorlayer.h"
+#include "qgsproject.h"
+#include "qgsexpressioncontextutils.h"
+#include "qgsmessagelog.h"
+#include <QAbstractTableModel>
+
+#include "qfield_core_export.h"
+
+OrderedRelationModel::OrderedRelationModel( QObject *parent )
+  : ReferencingFeatureListModel( parent )
+{
+}
+
+QString OrderedRelationModel::orderingField() const
+{
+  return mOrderingField;
+}
+
+void OrderedRelationModel::setOrderingField( const QString &orderingField )
+{
+  if ( mOrderingField == orderingField )
+    return;
+
+  mOrderingField = orderingField;
+  reload();
+  emit orderingFieldChanged();
+}
+
+QString OrderedRelationModel::imagePath() const
```

Comment:
Is there a specific reason why we wouldn't want image support for non-ordered relation model?

Reply:
It is not supported by QGIS, other than that it would be cool.